### ISSUE TITLE
ObjectNormalizer is not provided with serializer default_context

### ISSUE THREAD

**Issue by slava-petrov** [symfony/symfony#57316]
### Symfony version(s) affected

6.4.8

### Description

When ```default_context``` is set in framework.yaml globally - Symfony\Component\Serializer\Normalizer\ObjectNormalizer is missing them in specific circumstances. I have declared this in ```framework.yaml``` 

![image](https://github.com/symfony/symfony/assets/28899131/742667be-f794-412c-96e0-d7a13c5aa4f7)

```Symfony\Component\Serializer\DependencyInjection\SerializerPass.php``` does the binding 

https://github.com/symfony/symfony/blob/cf4ef922301b0906f94efa0a2ab0249885d8fc5c/src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php#L55

But after ```bin/console c:c``` ObjectNormalizer has an empty array passed to $defaultContext

`new \Symfony\Component\Serializer\Normalizer\ObjectNormalizer($c, $d, $b, $e, new \Symfony\Component\Serializer\Mapping\ClassDiscriminatorFromClassMetadata($c), NULL, [], $e)`

That's what is passed after cache clearing in ```App_KernelDevDebugContainer.php```.

If I manually enable autowiring in ```services.yaml``` like this:

![image](https://github.com/symfony/symfony/assets/28899131/eb37691d-bf6d-40a3-8cbc-12eb7437e2cc)

binding works and defaultContext is passed there. But looks like generally the issue is that bindings conflict with arguments setting somewhere during building process. Because, for example, ProblemNormalizer has $defaultContext passed correctly and it is not autowired by default as well.

My expectation is to have defaultContext passed there by default if it is configured globally for ObjectNormalizer

### How to reproduce

1. Setup clean symfony 6.4.8. ```symfony new 64lts --version="6.4.8"```
2. Install via composer serializer and property-access. ```composer require symfony/serializer```, ```composer require symfony/property-access```
3. Set ```dev``` environment and ```APP_DEBUG=true```
4. Add this to ```framework.yaml```
    ```
     serializer:
        default_context:
            disable_type_enforcement: true
            skip_null_values: true
4. ```bin/console c:c```

Check ObjectNormalizer declaration in container. $defaultContext is empty.

### Possible Solution

I see several options to solve this.

1. https://github.com/symfony/symfony/blob/cf4ef922301b0906f94efa0a2ab0249885d8fc5c/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L2004

Looks like on this line we already have $defaultContext variable containing empty array or globally set default_context if configured. And it is a bit weird that $defaultContext variable will be considered only if `circular_reference_handler` or `max_depth_handler` exist.

![image](https://github.com/symfony/symfony/assets/28899131/0b9027af-102d-429e-a206-d6dfbd436404)

And in case if these two options are missing - empty $context variable is set to ObjectNormalizer 6th argument. if we set $defaultContext here instead of declaring empty $context - it will work. 

```
        $defaultContext = $config['default_context'] ?? [];

        if ($defaultContext) {
            $container->setParameter('serializer.default_context', $defaultContext);
        }

        $arguments = $container->getDefinition('serializer.normalizer.object')->getArguments();

        if (isset($config['circular_reference_handler']) && $config['circular_reference_handler']) {
            $defaultContext += ($arguments[6] ?? $defaultContext) + ['circular_reference_handler' => new Reference($config['circular_reference_handler'])];
            $container->getDefinition('serializer.normalizer.object')->setArgument(5, null);
        }

        if ($config['max_depth_handler'] ?? false) {
            $defaultContext += ($arguments[6] ?? $defaultContext) + ['max_depth_handler' => new Reference($config['max_depth_handler'])];
        }

        $container->getDefinition('serializer.normalizer.object')->setArgument(6, $defaultContext);
```
2. Make ObjectNormalizer autowired by default

### Additional Context

_No response_

**Comment by HypeMC** [comment]
Duplicate of #56875.

Could you see if #57273 fixes your issue?

**Comment by slava-petrov** [comment]
@HypeMC , thanks for the fast reply

Yes, https://github.com/symfony/symfony/pull/57273 fixes my issue, it's actually really close to what I suggested above as a possible solution

Do you have any idea when this fix will be provided for `6.4` version?

I assume that currently I will have to add some workaround to make it work until this fix is released

**Comment by HypeMC** [comment]
> Do you have any idea when this fix will be provided for `6.4` version?

No, but bug fix versions are released approximately every month, so once the PR is merged, it shouldn't take long.

> I assume that currently I will have to add some workaround to make it work until this fix is released

You can stick to 6.4.7 until then, if I'm not mistaking, the bug is not present in that version.

**Comment by slava-petrov** [comment]
Thanks for a suggestion
